platform-core-module/src/main/java/com/softicar/platform/core/module/email/converter/EmailToPdfConverter.java

```diff
@@ -102,39 +129,98 @@ private boolean isInline(Part part) {
+	private void render(Part part) {
 
 		try {
-			String content = (String) part.getContent();
-			if (part.isMimeType(MimeType.TEXT_HTML.getIdentifier())) {
-				renderHtml(content, output);
-			} else if (part.isMimeType(MimeType.TEXT_PLAIN.getIdentifier())) {
-				var html = "<html><body><pre>" + escape(content) + "</pre></body></html>";
-				renderHtml(html, output);
+			Object content = part.getContent();
+
+			if (part.isMimeType("text/*")) {
+				String contentString = (String) content;
+				if (part.isMimeType(MimeType.TEXT_HTML.getIdentifier())) {
+					html.append(contentString);
+				} else if (part.isMimeType(MimeType.TEXT_PLAIN.getIdentifier())) {
+					html.append("<html><body><pre>" + escapePlainTextForHtml(contentString) + "</pre></body></html>");
+				}
+			}
+
+			else if (part.isMimeType("image/*")) {
+				String base64Data = getBase64Data((InputStream) content);
+				String mimeType = determineMimeType(part);
+				base64ImageMap.put(part.getFileName(), new Base64Image(mimeType, base64Data));
 			}
+
 		} catch (MessagingException | IOException exception) {
 			throw new RuntimeException(exception);
 		}
-	}
 
-	private String escape(String text) {
-
-		return text//
-			.replace("<", "&lt;")
-			.replace(">", "&gt;")
-			.replace("&", "&amp;");
 	}
 
 	private void renderHtml(String html, OutputStream output) {
 
-		var xhtml = convertHtmltoXhtml(html).replace("&nbsp;", " ");
-
+		var xhtml = convertHtmltoXhtml(html);
 		renderer.setDocumentFromString(xhtml);
 		renderer.layout();
 		renderer.createPDF(output, false);
 	}
 
-	private ByteArrayInputStream convertMsgToEml(Supplier<InputStream> input) {
+	private void assertEmlFormat(MimeMessage message) {
+
+		try {
+			if (message.getHeader("MIME-Version") == null) {
+				throw new RuntimeException("Expected a message in EML format but failed to find a 'MIME-Version' header.");
+			}
+		} catch (MessagingException exception) {
+			throw new RuntimeException(exception);
+		}
+	}
+
+	/**
+	 * Determines the mime type of the given {@link Part}.
+	 * <p>
+	 * Implementation note: In the return value of
+	 * {@link Part#getContentType()}, the mime type in {@code "foo/bar"} format
+	 * may or may not be followed by additional attributes, separated by either
+	 * a semicolon and/or various kinds and numbers of white spaces. Observed
+	 * examples include:
+	 *
+	 * <pre>
+	 * foo/bar
+	 * foo/bar name="filename.ext"
+	 * foo/bar;name="filename.ext"
+	 * foo/bar; name="filename.ext"
+	 * foo/bar;\nname="filename.ext"
+	 * foo/bar\nname="filename.ext"
+	 * foo/bar \nname="filename.ext"
+	 * </pre>
+	 *
+	 * @param part
+	 *            the {@link Part} for which the mime type shall be determined
+	 *            (never <i>null</i>)
+	 * @return the mime type in {@code "foo/bar"} format (never <i>null</i>)
+	 */
+	private static String determineMimeType(Part part) throws MessagingException {
+
+		return part.getContentType().replaceAll("\\s+", " ").replaceAll("[; ].*", "");
+	}
```

Comment:
I wonder if we should extract this into some `MimeTypeParser`...maybe in a follow-up PR?

Reply:
> Maybe we don't need this method at all?

Yes we do: Unfortunately, `jakarta.mail.Part` only provides `boolean isMimeType(String)` while _not_ providing something like `String getMimeType()`. We can only rely on `String getContentType()` which may or may not contain additional tokens after the sub-type.

See the Javadoc above this method, `determineMimeType`:
![image](https://user-images.githubusercontent.com/15086658/214780793-0d987d43-9c99-4d86-822e-a6c32b5172e1.png)

Reply:
We could certainly extract this as soon as there's a (re-)use case.